atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/atomic/AtomicUpdateResult.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.atomic;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.logsafe.Unsafe;
+import java.util.Collection;
+import org.immutables.value.Value;
+
+@Unsafe
+@Value.Immutable
+public interface AtomicUpdateResult {
+    ImmutableList<Cell> knownSuccessfullyCommittedKeys();
+
+    ImmutableList<Cell> existingKeys();
+
+    static AtomicUpdateResult create(Collection<Cell> committed, Collection<Cell> existing) {
```

Comment:
Hmm this would mean the layers above will be deciding which exception to throw with this data. 
This seems less than idea. I suppose I _could_ shove all of that logic in this class but that also does not seem like a great design.

Reply:
i suppose we can just have two different types of results as we have exceptions